xapian-core/api/registry.cc

```diff
@@ -72,6 +72,9 @@ class Xapian::Registry::Internal : public Xapian::Internal::intrusive_base {
     /// Clear all registered lat-long metrics.
     void clear_lat_long_metrics();
 
+    /// Clear all registered weighting schemes with short names.
+    void clear_weighting_schemes_short();
+
```

Comment:
This should just be done by the existing `clear_weighting_schemes()` - it doesn't make sense to clear the weighting schemes from one mapping but not the other.